subsys/net/lib/coap/coap.c

```diff
@@ -49,53 +49,132 @@ LOG_MODULE_REGISTER(net_coap, CONFIG_COAP_LOG_LEVEL);
 /* The CoAP message ID that is incremented each time coap_next_id() is called. */
 static uint16_t message_id;
 
-static inline bool append_u8(struct coap_packet *cpkt, uint8_t data)
+static int parse_option(uint8_t *data, uint16_t offset, uint16_t *pos, uint16_t max_len,
+			uint16_t *opt_delta, uint16_t *opt_len, struct coap_option *option);
```

Comment:
Is this forward declaration needed? I don't see any new use of this function before the actual definition.

Reply:
True. this is not needed anymore.